docs/src/modules/components/AppSearch.js

```diff
@@ -174,150 +71,429 @@ const Shortcut = styled('div')(({ theme }) => {
+              marginLeft: theme.spacing(0.5),
+              transition: 'margin 0.2s',
+              fontSize: theme.typography.pxToRem(16),
+            },
+            '& .DocSearch-Modal': {
+              boxShadow: `0px 4px 20px ${
+                theme.palette.mode === 'dark'
+                  ? alpha(theme.palette.background.paper, 0.7)
+                  : alpha(theme.palette.grey[700], 0.2)
+              }`,
+              ...(theme.palette.mode === 'dark' && {
+                border: '1px solid',
+                borderColor: theme.palette.primaryDark[700],
+              }),
+              borderRadius: theme.shape.borderRadius,
+            },
+            '& .DocSearch-SearchBar': {
+              borderBottom: '1px solid',
+              borderColor:
+                theme.palette.mode === 'dark'
+                  ? theme.palette.primaryDark[700]
+                  : theme.palette.grey[200],
+              padding: theme.spacing(1),
+            },
+            '& .DocSearch-Form': {
+              '& .DocSearch-Reset': {
+                display: 'none',
+              },
+              '& .DocSearch-Input': {
+                paddingLeft: theme.spacing(2.5),
+              },
+              '& .DocSearch-Search-Icon': {
+                width: '20px',
+                height: '20px',
+              },
+            },
+            '& .DocSearch-Cancel': {
+              display: 'block',
+              alignSelf: 'center',
+              height: '1.5rem',
+              marginRight: theme.spacing(1),
+              padding: theme.spacing(0.3, 0.8, 0.6, 0.8),
+              fontSize: 0,
+              borderRadius: 5,
+              backgroundColor:
+                theme.palette.mode === 'dark'
+                  ? theme.palette.primaryDark[800]
+                  : theme.palette.grey[50],
+              border: '1px solid',
+              borderColor:
+                theme.palette.mode === 'dark'
+                  ? theme.palette.primaryDark[600]
+                  : theme.palette.grey[300],
+              '&::before': {
+                content: '"esc"',
+                fontSize: theme.typography.pxToRem(13),
+                fontWeight: 700,
+                color: theme.palette.grey[600],
+              },
+            },
+            '& .DocSearch-Dropdown': {
+              minHeight: 384, // = StartScreen height, to prevent layout shift when first char
+              '&::-webkit-scrollbar-thumb': {
+                borderColor:
+                  theme.palette.mode === 'dark'
+                    ? theme.palette.primaryDark[900]
+                    : theme.palette.background.paper,
+                backgroundColor:
+                  theme.palette.mode === 'dark'
+                    ? theme.palette.primaryDark[700]
+                    : theme.palette.grey[500],
+              },
+              '&::-webkit-scrollbar-track': {
+                backgroundColor: theme.palette.background.paper,
+              },
```

Comment:
AFAIK, we could remove this, and use `colorScheme: theme.palette.mode` at the root 

```suggestion
```